.github/workflows/avro.yaml

```diff
@@ -0,0 +1,104 @@
+name: avro
+
+on:
+  push:
+    branches:
+      - master
+      - main
+    paths:
+      - '**.go'
+      - 'go.*'
+      - Makefile
+  pull_request:
+    paths:
+      - '**.go'
+      - 'go.*'
+      - Makefile
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+env:
+  directory: "."
+  allow_lint_failure: "true"
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up env
+        uses: actions/setup-go@v3
+        with:
+          cache: true
+          check-latest: true
+          cache-dependency-path: ${{ env.directory }}/go.sum
+          go-version-file: ${{ env.directory }}/go.mod
+      - name: Build
+        run: |
+          make build
+  test:
+    runs-on: ubuntu-22.04
+    container:
+      image: ghcr.io/heetch/containers/heetch-base-testenv:1
+    services:
+      registry:
+        image: lensesio/fast-data-dev:2.6.2-L0
+    steps:
+      - name: Purge initial context
+        run: |
+          rm -rf $HOME/* $HOME/.* || true
+      - name: Get BIN dir
+        id: get-bin-dir
+        shell: bash
+        run: |
+          echo bin_dir=.go-test-env >> $GITHUB_OUTPUT
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up env
+        uses: actions/setup-go@v3
+        with:
+          cache: true
+          check-latest: true
+          cache-dependency-path: ${{ env.directory }}/go.sum
+          go-version-file: ${{ env.directory }}/go.mod
+      - name: Install Tools
+        shell: bash
+        run: |
+          export GOBIN=${GITHUB_WORKSPACE}/${{ steps.get-bin-dir.outputs.bin_dir }}
+          go install github.com/mfridman/tparse@v0.12.1
```

Comment:
Use a pre-compiled binary instead if possible

Reply:
No you're right, we want to align with the version that is used in our private workflows, i.e. 0.12.1